constraint-solver/src/system_splitter.rs

```diff
@@ -33,22 +43,17 @@ pub fn split_system<T: RuntimeConstant, V: Clone + Ord + Hash + Display>(
                 }
             }
         }
-        constraint_system.retain_algebraic_constraints(|constr| {
```

Comment:
This was the main performance bottleneck, we iterate over the full list of constraints for each system we split out and thus have quadratic runtime.